### PR TITLE
run all commands with sudo after checking for sudo

### PR DESCRIPTION
**Pull request by sahil-lakhwani**
Signed-off-by: sahil-lakhwani <sahilakhwani@gmail.com>

Changes the LocalExecutor to run all commands with sudo
Checks if the current user is ableto sudo.

Have tested this with Ubuntu and CentOS

When running with non sudoer
```
./pf9ctl check-node
2021-02-24T11:22:26.2234Z	INFO	Loading configuration details
[sudo] password for dummy1: 
2021-02-24T11:22:28.6002Z	FATAL	Unable to perform pre-requisite checks on this node: User executing this CLI is not allowed to switch to privileged (sudo) mode
```

### CODE REVIEW

File: pkg/pmk/node.go [hunk 1]
```diff
@@ -234,3 +235,8 @@ func installHostAgentLegacy(ctx Config, auth keystone.KeystoneAuth, hostOS strin
 	zap.S().Info("Hostagent installed successfully")
 	return nil
 }
+
+func checkSudo(exec cmdexec.Executor) bool {
+	_, err := exec.RunWithStdout("bash", "-c", "echo '$(logname) ALL=(ALL:ALL) NOPASSWD: ALL' | (EDITOR='tee -a' visudo)")
```
Comment (anupbarve) [inline pkg/pmk/node.go:240]: What is `logname` ? I cannot find it's reference.

Comment (sahil-lakhwani) [reply pkg/pmk/node.go:240]: It's the login name I believe, the result of the command on a  machine with ubuntu user would be `sudo bash -c 'echo "$(logname) ALL=(ALL:ALL) NOPASSWD: ALL" | (EDITOR="tee -a" visudo)'`

But I think running `sudo -l` has the same behaviour and less chances of failure. Let me check and update the code if that is the case

Comment (sahil-lakhwani) [reply pkg/pmk/node.go]: Have update this with `sudo -l`

**Review by anupbarve — APPROVED** [review]
LGTM

**Pull request merged**